File: Bridge/System/Diagnostics/StackTrace.cs
using Bridge;
using System.Reflection;
using System.Text;
using System.Threading;

namespace System.Diagnostics
{
    public class StackTrace
    {
        internal enum TraceFormat
        {
            Normal,
            TrailingNewLine,
            NoResourceLookup
        }

        public const int METHODS_TO_SKIP = 0;
        
        public StackTrace()
        {
        }

        public StackTrace(bool fNeedFileInfo)
        {
        }

        public StackTrace(int skipFrames)
        {
        }

        public StackTrace(int skipFrames, bool fNeedFileInfo)
        {
        }
        
        public StackTrace(Exception e)
            : this(e, METHODS_TO_SKIP, false)
        {
        }

        public StackTrace(Exception e, bool fNeedFileInfo)
            : this(e, METHODS_TO_SKIP, fNeedFileInfo)
        {
        }

        public StackTrace(Exception e, int skipFrames)
            : this(e, skipFrames, false)
        {
        }

        public StackTrace(Exception e, int skipFrames, bool fNeedFileInfo)
        {
        }

        public StackTrace(StackFrame frame)
        {
        }

        public StackTrace(Thread targetThread, bool needFileInfo)
        {
        }

        internal StackTrace(StackFrame[] frames)
        {
        }

        public virtual int FrameCount
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public virtual StackFrame GetFrame(int index)
        {
            throw new NotImplementedException();
        }

        public virtual StackFrame[] GetFrames()
        {
            throw new NotImplementedException();
        }

        bool AddFrames(StringBuilder sb)
        {
            throw new NotImplementedException();
        }

        public static void GetFullNameForStackTrace(StringBuilder sb, MethodBase mi)
        {
            throw new NotImplementedException();
        }

        public override string ToString()
        {
            throw new NotImplementedException();
        }

        internal String ToString(TraceFormat traceFormat)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/diagnostics/stackTrace.js
    Bridge.define("System.Diagnostics.StackTrace", {
        statics: {
            fields: {
                METHODS_TO_SKIP: 0
            },
            ctors: {
                init: function () {
                    Object.defineProperty(this, "METHODS_TO_SKIP", {
                        get: function() {
                            return this._METHODS_TO_SKIP$ === undefined ? (this._METHODS_TO_SKIP$ = 0) : this._METHODS_TO_SKIP$;
                        },
                        set: function(v) {
                            this._METHODS_TO_SKIP$ = v;
                        }
                    });
                }
            },
            methods: {
                getFullNameForStackTrace: function (sb, mi) {
                    throw new System.NotImplementedException();
                }
            }
        },
        props: {
            FrameCount: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        ctors: {
            ctor: function () {
                this.$initialize();
            },
            $ctor1: function (fNeedFileInfo) {
                this.$initialize();
            },
            $ctor8: function (skipFrames) {
                this.$initialize();
            },
            $ctor9: function (skipFrames, fNeedFileInfo) {
                this.$initialize();
            },
            $ctor4: function (e) {
                System.Diagnostics.StackTrace.$ctor7.call(this, e, System.Diagnostics.StackTrace.METHODS_TO_SKIP, false);
            },
            $ctor5: function (e, fNeedFileInfo) {
                System.Diagnostics.StackTrace.$ctor7.call(this, e, System.Diagnostics.StackTrace.METHODS_TO_SKIP, fNeedFileInfo);
            },
            $ctor6: function (e, skipFrames) {
                System.Diagnostics.StackTrace.$ctor7.call(this, e, skipFrames, false);
            },
            $ctor7: function (e, skipFrames, fNeedFileInfo) {
                this.$initialize();
            },
            $ctor2: function (frame) {
                this.$initialize();
            },
            $ctor10: function (targetThread, needFileInfo) {
                this.$initialize();
            },
            $ctor3: function (frames) {
                this.$initialize();
            }
        },
        methods: {
            getFrame: function (index) {
                throw new System.NotImplementedException();
            },
            getFrames: function () {
                throw new System.NotImplementedException();
            },
            addFrames: function (sb) {
                throw new System.NotImplementedException();
            },
            toString: function () {
                throw new System.NotImplementedException();
            },
            toString$1: function (traceFormat) {
                throw new System.NotImplementedException();
            }
        }
    });
